piecrust-uplink/src/types.rs

```diff
@@ -134,3 +136,83 @@ impl core::fmt::Display for ContractId {
         Ok(())
     }
 }
+
+#[derive(Debug, Clone)]
+/// Result of the `raw` contract calls.
+pub struct RawResult {
+    pub data: Vec<u8>,
+    pub economic_mode: EconomicMode,
+}
+
+impl RawResult {
+    pub fn new(data: Vec<u8>, eco_mode: EconomicMode) -> Self {
+        Self {
+            data,
+            economic_mode: eco_mode,
+        }
+    }
+    pub fn empty() -> Self {
+        Self {
+            data: Vec::new(),
+            economic_mode: EconomicMode::None,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Archive, Deserialize, Serialize, Eq, PartialEq)]
+#[archive_attr(derive(CheckBytes))]
+pub enum EconomicMode {
+    Allowance(u64),
+    Charge(u64),
+    None,
+    Unknown,
+}
+
+impl EconomicMode {
+    const ALLOWANCE: u8 = 1;
+    const CHARGE: u8 = 2;
+    const NONE: u8 = 0;
+    const UNKNOWN: u8 = 3;
```

Comment:
That is exactly my opinion as well

Reply:
I think there's a misunderstanding here. I think the enum variant should still exist, I just think the tag is meaningless, since we should never really *write* it. Only when *reading*, when we don't understand what is on there, should we instantiate `EconomicMode::Unknown`, allowing the caller to then branch based on it not understanding what is in the buffer.

Reply:
you wrote: _I don't think it makes much sense to have a tag for EconomicMode::Unknown_, but you really meant `const UNKNOWN: u8 = 3;` right? So you are for keeping the EconomicMode::Unknown and for removing the const right?